test/connector/tcp/mssql/test

```diff
@@ -1,17 +1,126 @@
-#!/bin/bash
+#!/bin/bash -e
 
-export SECRETLESS_HOST=secretless
-if [[ ! -z $(docker-compose ps -q secretless-dev) ]]; then
-  export SECRETLESS_HOST=secretless-dev
+readonly db_host_tls="mssql"
+readonly db_host_no_tls="mssql" # TODO: configure a non-ssl container?
+readonly db_port=1433
+readonly db_user="sa"
+readonly db_password="yourStrong()Password"
+readonly secretless_port=2223
+readonly db_success_resp="1 rows affected"
+readonly db_fail_resp="Login failed for user"
+secretless_host="secretless"
+test_failed=false
+
+help_and_exit() {
+  local retval=${1:-1}
+  cat <<EOF
+This script tests access to an MSSQL server, first accessing the
+server directly (to establish a testing baseline), and then
+via a secretless broker.
+
+This script detects if you're in "dev" mode by the existence of
+the secretless-dev container. That container will exist only if
+the development process was started using the "./dev" command.
+
+Options:
+  -h Show this command usage
+EOF
+  exit "$retval"
+}
+
+while getopts "h" opt; do
+    case $opt in
+        h) help_and_exit 0;;
+        *) echo "Unknown option -$OPTARG"; help_and_exit 1;;
+    esac
+done
+
+contains_substring() {
```

Comment:
We should rename this to `assert_contains_substring` since the name `contains substring` suggest that it's a pure function that returns a boolean.  In fact, it's part of a little mini "test framework" we've built that checks an assertion, prints output, and alters the state of a global variable.

Reply:
Yes, that makes sense.